File: Linux/Custom Request 2/app/generate_flag.py
#!/usr/bin/python3

#|###############################|#
#| Chronos Security | Marin Radu |#
#| https://chronossec.site       |#
#| https://github.com/ChronosPK  |#
#|###############################|#

import hashlib, datetime, pytz

def generate_flag(salt):
	desired_timezone = pytz.timezone('Europe/Bucharest')
	current_time = datetime.datetime.now(desired_timezone).time()
	num_blocks_passed = (current_time.hour * 3600 + current_time.minute * 60 + current_time.second) // 300  # renew every 300 seconds

	flag = f"CSCTF{{}}{num_blocks_passed}{salt}"
	hashed_flag = hashlib.sha256(flag.encode()).hexdigest()
	flag = f"CSCTF{{{hashed_flag}}}"

	return flag

File: Web/Proxy man/app/generate_flag.php
<?php

#|###############################|#
#| Chronos Security | Marin Radu |#
#| https://chronossec.site       |#
#| https://github.com/ChronosPK  |#
#|###############################|#

function generate_flag($salt) {
    date_default_timezone_set('Europe/Bucharest');
    $current_time = time();
    $seconds_passed = date('H', $current_time) * 3600 + date('i', $current_time) * 60 + date('s', $current_time);
    $num_blocks_passed = floor($seconds_passed / 300);  // renew every 300 seconds

    $flag = "CSCTF{}" . $num_blocks_passed . $salt;
    $hashed_flag = hash("sha256", $flag);
    $flag = "CSCTF{" . $hashed_flag . "}";

    return $flag;
}

if(isset($_GET['generate_flag'])) {
    $salt = "Proxy man";
    $flag = generate_flag($salt);
    echo json_encode(['flag' => $flag]);
}
?>
